### PR TITLE
Update 'Configuring applications' to reflect actual django usage

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

File: docs/ref/applications.txt [hunk 1]
```diff
@@ -73,9 +73,8 @@ If there is no ``default_app_config``, Django uses the base
 ``django.contrib.admin`` to opt-in to :class:`~django.apps.AppConfig` features
 without requiring users to update their :setting:`INSTALLED_APPS`.
 
-New applications should avoid ``default_app_config``. Instead they should
```
Comment (graingert) [inline docs/ref/applications.txt]: does anyone know of any django apps on PyPI that actually force people to use 'package_name.apps.PackageNameAppConfig'?

Comment (graingert) [reply docs/ref/applications.txt]: the usual style is:
```
* PyPI: django-app-name
* INSTALLED_APPS: 'app_name'
```

**Comment by timgraham** [comment]
(Reverting 94a36cfd8c66a6be9a1e001b29e4c55071f4e0de).

**Comment by aaugustin** [comment]
This changes reinforces the special case of INSTALLED_APPS vs. most other settings referencing Python objects like MIDDLEWARE. It recommends a layer of implicit configuration and indirection — the lookup of `default_app_config` — which I find unpythonic.

I tried to do away with both of these things in the app loading refactor.

Honestly if we wanted to do something here, we should:

- deprecate `default_app_config` — I shouldn't have introduced it but I wanted to avoid another debate during the app loading refactor, as there were a bunch of other backwards incompatibilities already and many devs seem very attached to Django's magic around apps,
- alternatively, make it mandatory — I'll dislike that but at least we'll have everyone on the same page

I don't think it's very useful to simply change a weak recommendation hidden in the depths of the docs, as it'll only muddy the situation further.

**Comment by graingert** [comment]
@aaugustin it would be nice to know, what are the actual usage statistics of app.apps.AppAppConfig Vs app?

**Pull request closed without merging**